internal/server/routes.go

```diff
@@ -186,6 +189,21 @@ func add[Req, Res any](a *API, r *gin.RouterGroup, route route[Req, Res]) {
 	bindRoute(a, r, route.method, route.path, wrappedHandler)
 }
 
+func trimWhitespace(req interface{}) {
+	v := reflect.ValueOf(req)
+	for v.Type().Kind() == reflect.Pointer && !v.IsNil() {
+		v = v.Elem()
+	}
```

Comment:
Is this equivalent to https://pkg.go.dev/reflect#Indirect ? It looks similar, but `Indirect` doesn't have the nil check.